src/python/pants/engine/legacy/structs.py

```diff
@@ -349,34 +340,36 @@ def _filespec_for_exclude(raw_exclude, spec_path):
 
   @property
   @abstractmethod
-  def path_globs_kwarg(self):
+  def path_globs_kwarg(self) -> str:
     """The name of the `PathGlobs` parameter corresponding to this BaseGlobs instance."""
 
   @property
   @abstractmethod
-  def legacy_globs_class(self):
+  def legacy_globs_class(self) -> Type[wrapped_globs.FilesetRelPathWrapper]:
     """The corresponding `wrapped_globs` class for this BaseGlobs."""
 
-  def __init__(self, *patterns, **kwargs):
+  def __init__(
+    self, *patterns: str, spec_path: str, exclude: Optional[List[str]] = None, **kwargs,
```

Comment:
We were always looking for the kwargs `spec_path` and `exclude`, so I made them be explicit for better readability and to work with MyPy.

I kept `kwargs`, which should always be empty, so that we can keep our error message when users try to specify kwargs in BUILD files.